client/signup/test/utils.js

```diff
@@ -72,6 +73,73 @@ describe( 'utils', () => {
 		} );
 	} );
 
+	describe( 'getFilteredSteps', () => {
+		describe( 'when the given flow is found in the config', () => {
+			const exampleFlowName = 'onboarding';
+
+			describe( 'when there are a number of steps in the progress state', () => {
+				describe( 'and some of them match that flow', () => {
+					const userStep = { stepName: 'user' };
+					const siteTypeStep = { stepName: 'site-type' };
+					const someOtherStep = { stepName: 'some-other-step' };
+					const exampleSteps = {
+						user: userStep,
+						'site-type': siteTypeStep,
+						'some-other-step': someOtherStep,
+					};
+
+					const result = getFilteredSteps( exampleFlowName, exampleSteps );
+
+					test( 'it returns an array', () => {
+						expect( Array.isArray( result ) ).toBe( true );
+					} );
+
+					test( 'it should return only the step objects that match the flow', () => {
```

Comment:
Personally prefer writing these sort of object/array tests using `toEqual` which does a deep equality check.
```javascript
expect( result ).toEqual( [ userStep, siteTypeStep ] );
```

Using `toContain` has better semantics, which I like, but using `toEqual` will also test:
- elements are in the correct order
- there aren't other random elements in there
- doubles as an `isArray` test too :)